Lib/asyncio/runners.py

```diff
@@ -104,20 +102,7 @@ def run(self, coro, *, context=None):
         if context is None:
             context = self._context
 
-        if futures.isfuture(coro):
-            # This covers the argument being an asyncio.Future or asyncio.Task
-            task = coro
-        elif coroutines.iscoroutine(coro):
-            task = self._loop.create_task(coro, context=context)
-        elif inspect.isawaitable(coro):
-            async def _wrap_awaitable(awaitable):
-                return await awaitable
-
-            task = _wrap_awaitable(coro)
-        else:
-            raise ValueError("An asyncio Future, asyncio.Task, "
-                             "awaitable, or coroutine was expected, "
-                             "got {!r}".format(coro))
+        task = tasks.ensure_future(coro, loop=self._loop, context=context)
```

Comment:
I'm not sure that we should use `ensure_future` since the docs flag it as deprecated since 3.10.

Reply:
I don't think the function is deprecated. According to the docs, 3.10 only deprecates not specifying a loop explicitly when there's no currently running event loop.

In the specific case we're using it here, loop will always be specified explicitly, as it will be the loop associated with the `asyncio.Runner` instance that `run()` is being called on.